### DISCUSSION TITLE
User-defined gradient for each iteration of optimization

### DISCUSSION THREAD

**Discussion by YukiShirai** [casadi/casadi#3261] [Q&A]
Hello,

I'm wondering if the users can provide the gradients for each iteration of optimization. I know that there are some features for providing the gradients of the user-defined function according to [this](https://casadi.sourceforge.net/users_guide/html/node6.html).
This is a great feature but what I would like to do is slightly different. Suppose you have another module that computes gradients of the function with respect to the current value of the decision variables. Then, I use this module to get the gradient and I'd like to provide this gradient with the casadi.

Thanks for your attention and kind help in advance!

**Comment by jgillis** [comment]
There is a 'grad_f' option for nlpsol, but to implement it you'd need to do the user defined function.
